third_party/amd/lib/TritonAMDGPUTransforms/ReorderInstructions.cpp

```diff
@@ -236,6 +237,13 @@ static void scheduleGlobalLoadLocalStore(triton::FuncOp funcOp) {
       Block *defBlock = defOp->getBlock();
       if (!block->findAncestorOpInBlock(*defOp))
         return false;
+      // Don't hoist control flow as we don't track backtraces of ops within
+      // their regions.
+      if (isa<scf::IfOp, scf::ForOp, scf::WhileOp>(defOp)) {
```

Comment:
This could be `defOp->getNumRegions() != 0`. I think this could happen for `tt.reduce`?

Reply:
Well, it looks like `tl.reduce` can take a function with arbitrary code. This could include external constants? which could get out-of-order

Reply:
Actually I can't see a way to get that to fail. Just being overly cautious.

Reply:
I thought it had to be isolated from above but I guess I don't see anything blocking that